### DISCUSSION TITLE
hfs.delete(file) throws when file not exists

### DISCUSSION THREAD

**Discussion by magicdawn** [humanwhocodes/humanfs#104] [General]
hfs.delete(file) throws when file not exists

```txt
Error: ENOENT: no such file or directory, lstat
```

Other library behaviors
- fs-extra `fse.removeSync` silent when not exists

And deleting-files did not mention `when file not exists`
https://github.com/humanwhocodes/humanfs/blob/main/docs/file-operations.md#deleting-files

From the tips part, looks like read(json text bytes) / write / isFile / append allows none existing file. 
others delete / size / lastModified / move / copy did not mention.
but size / lastModified / move / copy semantically requires file existance. 

the rest `delete`, if it's designed to not allow delete none-exists file, a tip would be nice.

**Comment by magicdawn** [comment]
- https://github.com/humanwhocodes/humanfs/blob/node-v0.15.2/packages/node/src/node-hfs.js#L256
- https://nodejs.org/api/fs.html#fspromisesrmpath-options

use `fsp.rm(file, { force: true })` to ignore ENOENT error 🤣

**Comment by nzakas** [comment]
Thanks for the feedback. Yes, the docs aren't exhaustive at this point as I've still been trying to figure out how things should work.

I have been thinking about `delete()` for a while but haven't had time to circle back to it. I think I agree, it would be nice not to throw an error when the file is missing, but I'd also like to return a boolean indicating if an action was actually taken (similar to the `delete` operator in JS).
